examples/gas-in-box.py

```diff
@@ -0,0 +1,834 @@
+                    read_gmsh, filename=mesh_filename,
+                    mesh_construction_kwargs=mesh_construction_kwargs
+                )
+            else:
+                generate_mesh = partial(
+                    read_gmsh, filename=mesh_filename,
+                    mesh_construction_kwargs=mesh_construction_kwargs,
+                    force_ambient_dim=dim
+                )
+        else:
+            if dim is None:
+                dim = 2
+            nscale = max(nscale, 1)
+            scale_fac = pow(float(nscale), 1.0/dim)
+            nel_1d = int(scale_fac*24/dim)
+            from mirgecom.simutil import get_box_mesh
+            box_ll = -1
+            box_ur = 1
+            generate_mesh = partial(
+                get_box_mesh, dim=dim, a=(box_ll,)*dim, b=(box_ur,)*dim,
+                n=(nel_1d,)*dim, periodic=(periodic_mesh,)*dim,
+                tensor_product_elements=use_tpe)
+
+        local_mesh, global_nelements = distribute_mesh(comm, generate_mesh)
+        local_nelements = local_mesh.nelements
+        if dim is None:
+            dim = local_mesh.ambient_dim
+
+        def add_wonk(x: np.ndarray) -> np.ndarray:
+            wonk_field = np.empty_like(x)
+            if len(x) >= 2:
+                wonk_field[0] = (
+                    1.5*x[0] + np.cos(x[0])
+                    + 0.1*np.sin(10*x[1]))
+                wonk_field[1] = (
+                    0.05*np.cos(10*x[0])
+                    + 1.3*x[1] + np.sin(x[1]))
+            else:
+                wonk_field[0] = 1.5*x[0] + np.cos(x[0])
+
+            if len(x) >= 3:
+                wonk_field[2] = x[2] + np.sin(x[0] / 2) / 2
+            return wonk_field
+
+        if map_mesh:
+            from meshmode.mesh.processing import map_mesh
+            local_mesh = map_mesh(local_mesh, add_wonk)
+
+        if abs(rotation_angle) > 0:
+            from meshmode.mesh.processing import rotate_mesh_around_axis
+            theta = rotation_angle/180.0 * np.pi
+            local_mesh = rotate_mesh_around_axis(local_mesh, theta=theta)
+
+    dcoll = create_discretization_collection(actx, local_mesh, order=order,
+                                             tensor_product_elements=use_tpe)
+    nodes = actx.thaw(dcoll.nodes())
+    ones = dcoll.zeros(actx) + 1.
+
+    quadrature_tag = DISCR_TAG_QUAD if use_overintegration else None
+
+    vis_timer = None
+
+    if logmgr:
+        logmgr_add_cl_device_info(logmgr, queue)
+        logmgr_add_device_memory_usage(logmgr, queue)
+
+        vis_timer = IntervalTimer("t_vis", "Time spent visualizing")
+        logmgr.add_quantity(vis_timer)
+
+        logmgr.add_watches([
+            ("step.max", "step = {value}, "),
+            ("t_sim.max", "sim time: {value:1.6e} s\n"),
+            ("t_step.max", "------- step walltime: {value:6g} s, "),
+            ("t_log.max", "log walltime: {value:6g} s")
+        ])
+
+    velocity = np.zeros(shape=(dim,))
+    species_diffusivity = None
```

Comment:
Move these 3 lines down to keep all transport stuff together.

Reply:
I just need the velocity to stay put.